assets/js/components/KeyMetrics/hooks/useChangeMetricsFeatureTourEffect.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Change metrics feature tour hook.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+import sharedKeyMetrics from '../../../feature-tours/shared-key-metrics';
+const { useSelect, useDispatch } = Data;
+
+/**
+ * Triggers on demand tour for shared key metrics if all conditions are met.
+ *
+ * @since n.e.x.t
+ *
+ * @param {boolean} renderChangeMetricLink If metric link meets the conditions to render.
+ */
+export const useChangeMetricsFeatureTourEffect = ( renderChangeMetricLink ) => {
```

Comment:
@eugene-manuilov It wasn't required in IB, it was a suggestion on previous CR, it was originally in `ChangeMetricsLink`, but since it was adding certain number of lines for this logic, it was moved to separate hook - comment for reference - https://github.com/google/site-kit-wp/pull/7661#discussion_r1346323256

Let me know what you think

Reply:
Ah... okay, then let's keep it as is although that is less appropriate IMO because:

1. It makes us have more files in the codebase than we really need.
1. It makes the logic in the component harder to read and understand because we need to find and open an additional file to understand what happens in that hook.
1. Webpack will bundle component and hook as separate modules thus the size of the final bundle will be slightly bigger than it could be if we not extract that logic into a custom hook.

cc @aaemnnosttv

Reply:
Thanks for your review @eugene-manuilov ! A few thoughts in response to your comments above.

> * It makes us have more files in the codebase than we really need.

We shouldn't have more files for no reason of course, but we often use more files to keep each of them smaller and easier to understand and extract complexity. I don't think that it only makes sense to extract components/hooks if there is more than one usage of them, there are plenty of instances where we extract parts to keep the overall size and complexity down, e.g. `BannerNotification` components.

> * It makes the logic in the component harder to read and understand because we need to find and open an additional file to understand what happens in that hook.

I find this point a bit hard to see, as I feel that it's harder to understand the component with a large portion of its code only needed for the tour effect which is itself easier to understand in isolation IMO. In the context of the component, we see that it has a named effect about its tour, so you could argue that it's easier to understand in this way.

> * Webpack will bundle component and hook as separate modules thus the size of the final bundle will be slightly bigger than it could be if we not extract that logic into a custom hook.

This is true, but the increase in size is probably only a handful of bytes which isn't a concern. There are likely many other things we could optimize to reduce the overall size of our assets in more impactful ways, so I don't think bundle size is an issue here. We have the compressed size action to help highlight changes to our bundle size which reported didn't show any substantial change here.